### ISSUE TITLE
Specify a local npm registry to be used with npmgraph

### ISSUE THREAD

**Issue by blured75** [npmgraph/npmgraph#19]
Hi, 
Thanks a lot for your awesome work !
Is it possible to specify a local npm server like verdaccio to be used with npm graph ?
Regards,
Blured.

**Comment by robross0606** [comment]
I have the same or similar question.  The running instance seems to ignore my `npmrc` settings which define a scoped private repository.  Is there a way to configure a running instance to use something other than the global public repo?

**Comment by broofa** [comment]
Briefly, no, this isn't something that has been designed for, but it's probably doable.

The main issues, off the top of my head:
1. Make the repository host configurable.  (currently uses `registry.npmjs.cf`)  Probably not a big deal.  Add a UI to the "ⓘ" tab for this, store in `localStorage`, etc.
2. Make secondary hosts configurable.  E.g. `npmjs.com` for links to module info
3. Figure out how to whitelist private hosts [in the Content-Security-Policy](https://github.com/npmgraph/npmgraph/blob/main/index.html#L6).
3. Handle private modules not being available in services like `npms.io` and `bundlephobia.com`

**Comment by robross0606** [comment]
@broofa npm already has a mechanism for doing this via npmrc environment.  The trickiest part here is that internal repositories may be configured specifically for one or more "scoped" packages, while all others are still sourced from the public repo.  Reproducing this flexibility using other mechanisms like .env seems like it would be difficult.  Would it be possible to have an `.npmrc` deployed on the server and use that somehow?  I already see one in project source.  Is that already referenced in the same ways it is for regular `npm` usage?

**Comment by broofa** [comment]
> Reproducing this flexibility using other mechanisms like UI or .env seems like it would be difficult.

Fair enough.  Using this project's .npmrc file is likely to end up being problematic, however, as it's purpose is to provide configuration specific to _this _project_.    So maybe have a command-line flag or env var for specifying a path to whatever .npmrc file the user has for their environment...?

**Comment by broofa** [comment]
Quick update for @EmmetBeachBum (et al)...

I've never used a private registry, so this feature isn't high on my personal list of needs/wants for this project.  I'd be open to a PR that added support, however.  ('Would have to include instructions for setting up a local Verdaccio repo so I could test, though.)
